### PR TITLE
[14.0][FIX] web_field_required_invisible_manager: reset on change  + remake concept

### PR DESCRIPTION
**Pull request by ilyasProgrammer**

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional review ok! Previous version was not working as described. 

@pedrobaeza asking your opinion since you merged the original #2502 , thanks!

**Comment by pedrobaeza** [comment]
Well, I can't say, as I merged it due to the existing reviews and I'm not using this module.

File: web_field_required_invisible_manager/models/custom_field_restriction.py [hunk 1]
```diff
@@ -41,18 +44,73 @@ class CustomFieldRestriction(models.Model):
...
     field_invisible = fields.Boolean()
+    field_readonly = fields.Boolean()
+    # generated technical fields used in form attrs:
+    visibility_field_id = fields.Many2one("ir.model.fields", ondelete="cascade")
+    readonly_field_id = fields.Many2one("ir.model.fields", ondelete="cascade")
+    required_field_id = fields.Many2one("ir.model.fields", ondelete="cascade")
 
     @api.onchange("field_id")
     def onchange_field_id(self):
         self.required = self.field_id.required
```
Comment (geomer198) [inline web_field_required_invisible_manager/models/custom_field_restriction.py]: Could you use `update()` function for updating values.

**Review by geomer198 — APPROVED** [review]
LGTM!

**Pull request merged**